lib/ConnectionContextBase.ts

```diff
@@ -168,11 +168,13 @@ export module ConnectionContextBase {
       const socket = parameters.config.webSocket || (window as any).WebSocket;
       const host = parameters.config.host;
       const endpoint = parameters.config.webSocketEndpointPath || "";
+      const socketOptions = parameters.config.webSocketConstructorOptions || "";
```

Comment:
Is empty string the right default here?

Reply:
Should probably be {}.

Reply:
Empty string works fine here but if I default to "", then the type is string | Options. 

Which is not what we want, I'll just init to an empty object.